core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -183,6 +300,21 @@ private Object invoke(ServiceRequestContext ctx, HttpRequest req, @Nullable Aggr
     }
 
     /**
+     * Whether the request has form data.
+     */
+    private static boolean hasFormData(HttpRequest req) {
+        switch (req.method()) {
+            case POST:
+            case PUT:
+                String contentType = req.headers().get(HttpHeaderNames.CONTENT_TYPE);
+                return contentType != null &&
+                       MediaType.FORM_DATA.toString().equals(HttpUtil.getMimeType(contentType));
```

Comment:
Never mind. It seems like `HttpUtil.getMimeType()` returns without media type attributes which is fine for this case. We would need to extract the `charset` attribute later though?